src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -523,6 +531,23 @@ JNI_METHOD(jboolean, isConnected)(JNIEnv * env, jobject self, jlong deviceContro
     return JNI_FALSE;
 }
 
+JNI_METHOD(jstring, getIpAddress)(JNIEnv * env, jobject self, jlong deviceControllerPtr)
+{
+    ChipDeviceController * deviceController = (ChipDeviceController *) deviceControllerPtr;
+
+    chip::Inet::IPAddress addr;
+    char addrStr[50];
+
+    {
+        ScopedPthreadLock lock(&sStackLock);
+        if (!deviceController->GetIpAddress(addr))
+            return nullptr;
```

Comment:
will this throw an exception or return null?
I wonder if it would be better to throw instead of having getIpAddress() potentially be able to return null.

Reply:
return null which indicates that the address hasn't been set or received from the peer device yet. It's a valid situation and I would need to try/catch in the caller if the function threw, so I'm not sure if we should do that...